package.json

```diff
@@ -18,19 +18,20 @@
   "types": "lib/peg.d.ts",
   "bin": "bin/peggy",
   "scripts": {
-    "pretest": "npm run parser",
-    "test": "npm run test:bare",
-    "test:bare": "mocha --recursive test --exclude test/browser.stub.js --reporter min",
+    "clean": "rimraf -f build && rimraf -f browser && mkdir build && cd build && mkdir ts && mkdir rollup && mkdir terser && cd .. && mkdir browser",
```

Comment:
windows doesn't have `rm`, and maybe windows supports mkdir, but not -p?

Reply:
> What is purpose of rimraf? Isn't simple rm -rf enough

No.  This is a famous problem with `node` that's been roiling the standard library for three majors now.  

&nbsp;

> Also is manually creating directories is really necessary? 

Yes.  If you don't, the build fails.  Try taking it out and running it.

&nbsp;

> I think, that all tools are able to create necessary directories during their work

Just try it and see

Then look in the build notes for current node, where they've moved this through four functions in the standard library and changed the API every time

This is sort of standard experienced node programmer stuff.  If you expect to support older node, you need to install small packages to cover over for the holes in the node standard library.

&nbsp;

> Even if you create directories manually, the mkdir have the -p flag that allow you to create all directories on the path

That won't work on Windows, older Macintoshes, or things not running `bash/zsh`

Node is a portable community.  Relying on the things that exist in your Windows shell is considered unacceptable.

Reply:
> windows doesn't have rm, and maybe windows supports mkdir, but not -p?

When in DOS shell, windows doesn't have rm 

When in DOS shell, the windows mkdir api doesn't fit unix expectations

No version of windows shell supports -p (powershell can be told to do it anyway by sending erroraction=0, but that's dangerous, because that'll make it do other things you don't want, too, like steamrolling symbolic links)

Reply:
I'm just surprising why you not use one tool everywhere. If `rimraf` is necessary to outcome some problems, well, no problem, just use it in every place. Otherwise I don't understand why the `deploy` task still uses `rm` that potentially missed on Windows.

Reply:
> I'm just surprising why you not use one tool everywhere. If rimraf is necessary to outcome some problems, well, no problem, just use it in every place. Otherwise I don't understand why the deploy task still uses rm that potentially missed on Windows.

Because `rm` is faster.  `rimraf` is involved because it deletes trees.

It's like when you told me to remove the star from the glob in `.npmignore`.  It's not important.  It's a largely meaningless speed detail that we get "right" because it's fun

Reply:
Oh, because the contrast is that now that it's in `npm scripts` ***it's not Windows interpreting the command first, it's npm - it gets parsed before it's forked out, because of path stuff - and rm is specifically shimmed in, but doesn't have tree support***

Sorry, I tend to forget to mention that

`npm` does some weird stuff in the name of portability.  Some of that is what a web programmer would call an attempt to shim the unix api.